packages/framework/tree-agent/src/prompt.ts

```diff
@@ -121,39 +121,44 @@ ${getTreeMapNodeDocumentation(mapInterfaceName)}
 `;
 
 	const rootTypes = normalizeFieldSchema(schema).allowedTypeSet;
-	const prompt = `You are a helpful assistant collaborating with the user on a document. The document state is a JSON tree, and you are able to analyze and edit it.
-The JSON tree adheres to the following Typescript schema:
-
-\`\`\`typescript
-${typescriptSchemaTypes}
-\`\`\`
-
-If the user asks you a question about the tree, you should inspect the state of the tree and answer the question. When answering such a question, DO NOT answer with information that is not part of the document unless requested to do so.
-If the user asks you to edit the tree, you should use the "${editingToolName}" tool to accomplish the user-specified goal, following the instructions for editing detailed below.
+	const editing = `If the user asks you to edit the tree, you should author a JavaScript function to accomplish the user-specified goal, following the instructions for editing detailed below.
+${editToolName === undefined ? "Use an applicable tool to perform the edit (if one is available)." : `You must use the "${editToolName}" tool to perform the edit.`}
```

Comment:
Why do we support a world in which there isn't an edit tool?

Reply:
See my other comment, but I don't view this as any different than having test logic in production code.